packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -252,31 +255,46 @@ const DataGrid = forwardRef((props, ref) => {
       columnId: displayedColumns[newColIndex].id,
     };
 
-    if (makeActiveElement) {
-      let focusedCell;
-      if (isSection(newRowIndex)) {
-        [focusedCell] = grid.current.rows[newRowIndex].cells;
+    if (!makeActiveElement) {
+      return;
+    }
 
-        if (!focusedCell.hasAttribute('tabindex')) {
-          focusedCell = grid.current.rows[newRowIndex].querySelector('button');
-        }
-      } else {
-        // Set focus on input field (checkbox) of row selection cells.
-        focusedCell = grid.current.rows[newRowIndex].cells[newColIndex];
-        if (isRowSelectionCell(newColIndex) && focusedCell.getElementsByTagName('input').length > 0) {
-          [focusedCell] = focusedCell.getElementsByTagName('input');
-        }
+    let focusedCell;
 
-        // Set focus to column header button, if it exists
-        const isHeaderRow = (newRowIndex === 0 || (hasColumnHeaderActions && newRowIndex === 1));
-        if (isHeaderRow && !focusedCell.hasAttribute('tabindex')) {
-          focusedCell = focusedCell.querySelector('[role="button"]') || focusedCell.querySelector('button');
-        }
+    if (isSection(newRowIndex)) {
+      [focusedCell] = grid.current.rows[newRowIndex].cells;
+
+      if (!focusedCell.hasAttribute('tabindex')) {
+        focusedCell = grid.current.rows[newRowIndex].querySelector('button');
       }
+      focusedCell?.focus();
+      return;
+    }
+
+    focusedCell = grid.current.rows[newRowIndex].cells[newColIndex];
 
+    // If there are multiple focusable elements, set focus on the cell
+    if (getFocusableElements(focusedCell).length > 1) {
       focusedCell?.focus();
+      return;
+    }
+
+    // Set focus on input field (checkbox) of row selection cells.
+    if (isRowSelectionCell(newColIndex) && focusedCell.getElementsByTagName('input').length === 1) {
+      [focusedCell] = focusedCell.getElementsByTagName('input');
+      focusedCell?.focus();
+      return;
+    }
+
+    // Set focus to a single header button or hyperlink if they are the only content in cell
+    if (!focusedCell.hasAttribute('tabindex') || focusedCell.getElementsByTagName('button').length === 1 || focusedCell.getElementsByTagName('a').length === 1) {
+      focusedCell = focusedCell.querySelector('[role="button"]') || focusedCell.querySelector('button') || focusedCell.querySelector('a');
```

Comment:
If you are using a querySelector, you could use a single selector instead of having 3 separate conditions to query the DOM.  You can use commas in your selector to separate.

Reply:
Updated here: d86a31695501e5e103ba9f36adff2a349fedc074